docs/topics/security.txt

```diff
@@ -172,18 +210,98 @@ vulnerability allows and much worse).
 
 Subdomains within a site will be able to set cookies on the client for the whole
-domain.  By setting the cookie and using a corresponding token, subdomains will
-be able to circumvent the CSRF protection.  The only way to avoid this is to
+domain. By setting the cookie and using a corresponding token, subdomains will
+be able to circumvent the CSRF protection. The only way to avoid this is to
 ensure that subdomains are controlled by trusted users (or, are at least unable
-to set cookies).  Note that even without CSRF, there are other vulnerabilities,
-such as session fixation, that make giving subdomains to untrusted parties a bad
-idea, and these vulnerabilities cannot easily be fixed with current browsers.
+to set cookies).
+
+Note that even without CSRF, there are other vulnerabilities, such as session
+fixation, that make giving subdomains to untrusted parties a bad idea, and
+these vulnerabilities cannot easily be fixed with current browsers.
+
+CSRF protection design policy
+-----------------------------
+
+CSRF token pairs (cookie and POST data) are:
+
+* arbitrary
+* not linked to sessions by default.
+
+Arbitrary CSRF token pairs
+~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This will occasionally be flagged in security audits as a vulnerability. That
+is, when Django checks the cookie against the POSTed hidden field, it only
+checks that they match *each other*, not that they match the server secret on
+which they are both based, or that they can be verified against the user's
+session.
+
+This means an attacker could assemble a POST request on a victim's machine,
+containing a matching cookie-field pair, that would pass the CSRF checks.
+
+It's possible to stage a 'proof' of such an attack by manipulating one's own
+cookies and constructing malicious POST requests that defeat CSRF protection.
+This would be akin to complaining of a weakness in one's own front-door by
+demonstrating that it can be opened by someone inside the house - in each case,
+an attack of this kind is predicated on a high level of access already.
+
+In a real CSRF attack, the attacker would need access to the victim's cookies
+to be able to achieve this, via (for example) a man-in-the-middle or cross-site
+scripting attack, in which case the victim has already been thoroughly
+compromised, and exploiting CSRF is almost certainly an unnecessary extra
+effort.
+
+CSRF tokens not associated with sessions by default
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+CSRF tokens *can* be linked to sessions by using the
+:setting:`CSRF_USE_SESSIONS`, which stores them in the user's session cookie.
+(note that this would provide no additional protection against the
+CSRF-defeating attacks described above).
+
+However this is not the default, because *not* linking CSRF protection to a
+session allows using the protection on sites (for example, pastebins such as
+`dpaste.com <http://dpaste.com>`_ that allow submissions from anonymous users
+who don't have a session.
```

Comment:
> protection on sites (for example, pastebins such as`dpaste.com <http://dpaste.com>`_ that allow submissions from anonymous users who don't have a session.

protection on sites that allow submissions from anonymous users who don't have a session, such as pastebins like `dpaste.com <http://dpaste.com>`_.

(For those playing at home, dpaste is used by `templates/technical_500.html`. TIL.)